docs/guides/unity-catalog.md

```diff
@@ -1,12 +1,132 @@
+- [Initialize the required providers](#provider-initialization)
+- [Configure AWS objects](#configure-aws-objects)
+  - A S3 bucket to store data from managed tables in Unity Catalog
+  - An IAM policy to define permissions to access data
+  - An IAM role that will be assumed by Unity Catalog to access data 
+- [Create users and groups who can access Unity Catalog](#create-users-and-groups)
+- [Create a Unity Catalog metastore and link it to workspaces](#create-a-unity-catalog-metastore)
+- [Create Unity Catalog objects in the metastore](#create-unity-catalog-objects)
+- [Configure external tables and credentials](#configure-external-tables)
+
+## Provider initialization
+
+> Initialize provider in "MWS" mode to set up account-level resources
+
 ```hcl
+terraform {
+  required_providers {
+    databricks = {
+      source  = "databrickslabs/databricks"
+      version = "0.4.6"
+    }
+    aws = {
+      source  = "hashicorp/aws"
+      version = "3.49.0"
+    }
+  }
+}
+
 provider "aws" {
-  region = local.region
+  region = var.region
 }
 
+// initialize provider in "MWS" mode for account-level resources
+provider "databricks" {
+  alias      = "mws"
+  host       = "https://accounts.cloud.databricks.com"
+  account_id = var.databricks_account_id
+  username   = var.databricks_account_username
+  password   = var.databricks_account_password
+}
+
+// initialize provider at workspace level, to create UC resources
+provider "databricks" {
+  alias    = "workspace"
+  host     = var.databricks_workspace_host
+  username = var.databricks_account_username
+  password = var.databricks_account_password
+}
+```
+
+Define the required variables
+
+```hcl
+variable "databricks_account_username" {}
+variable "databricks_account_password" {}
+variable "databricks_account_id" {}
+variable "databricks_workspace_host" {}
+
+variable "tags" {
+  default = {}
+}
+
+variable "region" {
+  default = "eu-west-1"
+}
+
+variable "databricks_workspace_ids" {
+  description = <<EOT
+  List of Databricks workspace ids to be enabled with Unity Catalog.
+  Enter with square brackets and double quotes
+  e.g. ["111111111", "222222222"]
+  EOT
+  type        = list(string)
+}
+
+variable "databricks_users" {
+  description = <<EOT
+  List of Databricks users to be added at account-level for Unity Catalog.
+  Enter with square brackets and double quotes
+  e.g ["first.last@domain.com", "second.last@domain.com"]
+  EOT
+  type        = list(string)
+}
+
+variable "databricks_unity_admins" {
+  description = <<EOT
+  List of Admins to be added at account-level for Unity Catalog.
+  Enter with square brackets and double quotes
+  e.g ["first.admin@domain.com", "second.admin@domain.com"]
+  EOT
+  type        = list(string)
+}
+
+variable "unity_admin_group" {
+  description = "Name of the admin group. This group will be set as the owner of the Unity Catalog metastore"
+  type        = string
+}
+
+resource "random_string" "naming" {
```

Comment:
add a comment why we might need a random naming

Reply:
done